lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -457,6 +457,96 @@ def all_whitelisted_configurations
 
       #--------------------------------------#
 
+      def raw_use_modular_headers_hash
+        get_hash_value('use_modular_headers', {})
+      end
+      private :raw_use_modular_headers_hash
+
+      # Returns the use_modular_headers hash pre-populated with default values.
+      #
+      # @return [Hash<String, Array>] Hash with :all key for inhibiting all
+      #         warnings, :for_pods key for inhibiting warnings per Pod,
+      #         and :not_for_pods key for not inhibiting warnings per Pod.
+      #
+      def use_modular_headers_hash
+        raw_hash = raw_use_modular_headers_hash
+        if exclusive?
+          raw_hash
+        else
+          parent_hash = parent.send(:use_modular_headers_hash).dup
+          if parent_hash['not_for_pods']
+            # Remove pods that are set to not inhibit inside parent if they are set to inhibit inside current target.
+            parent_hash['not_for_pods'] -= Array(raw_hash['for_pods'])
+          end
+          if parent_hash['for_pods']
+            # Remove pods that are set to inhibit inside parent if they are set to not inhibit inside current target.
+            parent_hash['for_pods'] -= Array(raw_hash['for_pods'])
+          end
+          if raw_hash['all']
+            # Clean pods that are set to not inhibit inside parent if inhibit_all_warnings! was set.
+            parent_hash['not_for_pods'] = nil
+          end
+          parent_hash.merge(raw_hash) do |_, l, r|
+            Array(l).concat(r).uniq
+          end
+        end
+      end
+
+      # @return [Bool] whether the target definition should inhibit warnings
+      #         for a single pod. If inhibit_all_warnings is true, it will
+      #         return true for any asked pod.
+      #
+      def build_pod_as_module?(pod_name)
```

Comment:
is this more like a `build_module_for_pod?(pod_name)`?

Reply:
not really